mlflow/pyfunc/__init__.py

```diff
@@ -411,8 +411,13 @@ def predict(self, data: PyFuncInput) -> PyFuncOutput:
         :return: Model predictions as one of pandas.DataFrame, pandas.Series, numpy.ndarray or list.
         """
         input_schema = self.metadata.get_input_schema()
+        _logger.debug("Data before _enforce_schema: %s", data)
+        _logger.debug("Input schema: %s", input_schema)
+        # Input schema: ['input_documents': string]
+        # Expected: ['input_documents': string (Document?), 'question': string]
```

Comment:
```
2023/05/23 11:05:37 DEBUG mlflow.pyfunc: Data before _enforce_schema: [{'input_documents': [Document(page_content='I love MLflow.', metadata={'source': '/path/to/mlflow.txt'}), Document(page_content='I love langchain.', metadata={'source': '/path/to/langchain.txt'}), Document(page_content='I love AI.', metadata={'source': '/path/to/ai.txt'})], 'question': 'What do I like?'}]
2023/05/23 11:05:37 DEBUG mlflow.pyfunc: Input schema: ['input_documents': string]
2023/05/23 11:05:37 DEBUG mlflow.pyfunc: Data after _enforce_schema:                                      input_documents
0  [page_content='I love MLflow.' metadata={'sour...
2023/05/23 11:05:37 DEBUG mlflow: Input pandas data:                                                                                                                                                                                                                 input_documents
0  [page_content='I love MLflow.' metadata={'source': '/path/to/mlflow.txt'}, page_content='I love langchain.' metadata={'source': '/path/to/langchain.txt'}, page_content='I love AI.' metadata={'source': '/path/to/ai.txt'}]
2023/05/23 11:05:37 DEBUG mlflow: messages = [{'input_documents': [Document(page_content='I love MLflow.', metadata={'source': '/path/to/mlflow.txt'}), Document(page_content='I love langchain.', metadata={'source': '/path/to/langchain.txt'}), Document(page_content='I love AI.', metadata={'source': '/path/to/ai.txt'})]}]
2023/05/23 11:05:37 DEBUG mlflow.langchain.api_request_parallel_processor: requests = [{'input_documents': [Document(page_content='I love MLflow.', metadata={'source': '/path/to/mlflow.txt'}), Document(page_content='I love langchain.', metadata={'source': '/path/to/langchain.txt'}), Document(page_content='I love AI.', metadata={'source': '/path/to/ai.txt'})]}]
2023/05/23 11:05:37 DEBUG mlflow: Input pandas data:                                                                                                                                                                                                                 input_documents
0  [page_content='I love MLflow.' metadata={'source': '/path/to/mlflow.txt'}, page_content='I love langchain.' metadata={'source': '/path/to/langchain.txt'}, page_content='I love AI.' metadata={'source': '/path/to/ai.txt'}]
2023/05/23 11:05:37 DEBUG mlflow: messages = [{'input_documents': [Document(page_content='I love MLflow.', metadata={'source': '/path/to/mlflow.txt'}), Document(page_content='I love langchain.', metadata={'source': '/path/to/langchain.txt'}), Document(page_content='I love AI.', metadata={'source': '/path/to/ai.txt'})]}]
2023/05/23 11:05:37 DEBUG mlflow.langchain.api_request_parallel_processor: requests = [{'input_documents': [Document(page_content='I love MLflow.', metadata={'source': '/path/to/mlflow.txt'}), Document(page_content='I love langchain.', metadata={'source': '/path/to/langchain.txt'}), Document(page_content='I love AI.', metadata={'source': '/path/to/ai.txt'})]}]
```

Reply:
It turns out to be a langchain issue. The `chain.input_keys` does not return the correct value.